packages/world/src/systems/BuildSystem.sol

```diff
@@ -28,18 +28,17 @@ import { BUILD_ENERGY_COST } from "../Constants.sol";
 import { EntityId } from "../types/EntityId.sol";
 import { ObjectType, ObjectTypes } from "../types/ObjectType.sol";
 
-import "../ProgramHooks.sol" as Hooks;
 import { ProgramId } from "../types/ProgramId.sol";
 import { Orientation, Vec3, vec3 } from "../types/Vec3.sol";
 
 struct BuildContext {
   EntityId caller;
+  ObjectType slotType;
+  ObjectType buildType;
```

Comment:
why is there both slot and build type? arent they always the same?

Reply:
btw this was here previously, I think I added it when trying to implement the water placement stuff (slot type is water bucket in this case).

Reply:
maybe we pass in `SlotData` then

Reply:
wdym? to the hook?